File: Shared/src/main/java/nju/classroomassistant/shared/messages/login/LoginMessage.java
package nju.classroomassistant.shared.messages.login;

import nju.classroomassistant.shared.messages.Message;

public class LoginMessage extends Message {
    private String studentId;

    public LoginMessage(String studentId) {
        this.studentId = studentId;
    }

    public String getStudentId() {
        return studentId;
    }

    public void setStudentId(String studentId) {
        this.studentId = studentId;
    }
}

File: shared/src/main/kotlin/nju/classroomassistant/shared/messages/login/LoginMessage.kt
package nju.classroomassistant.shared.messages.login

import nju.classroomassistant.shared.messages.Message
import java.io.Serializable

class LoginMessage(
    val studentId: String
): Message()

enum class LoginResponse: Serializable {
    OK,
    NOT_ALLOWED, // 没有选这个课
    ERROR // 出错了
}

class LoginResponseMessage(
    val response: LoginResponse
): Message()